diff -up amanda-2.6.1p2/common-src/glib-util.c.ftbfs amanda-2.6.1p2/common-src/glib-util.c
--- amanda-2.6.1p2/common-src/glib-util.c.ftbfs	2010-02-16 13:06:25.000000000 +0100
+++ amanda-2.6.1p2/common-src/glib-util.c	2010-02-16 13:06:52.000000000 +0100
@@ -45,7 +45,7 @@ glib_init(void) {
      * is initialized) */
 #ifdef HAVE_LIBCURL
 # ifdef G_THREADS_ENABLED
-    g_assert(!g_thread_supported());
+/*   g_assert(!g_thread_supported()); */
 # endif
     g_assert(curl_global_init(CURL_GLOBAL_ALL) == 0);
 #endif
